### PR TITLE
add I3 support + update to formspecs v2

### PR DESCRIPTION
**Pull request by wsor4035**
what the title says

### CODE REVIEW

**Comment by fluxionary** [comment]
Fixing luacheck by making it assume the expected fields are present, is acceptable to me

**Review by BuckarooBanzay — APPROVED** [review]

**Comment by fluxionary** [comment]
Should you also add i3 stylization to the filter injector formspecs?

**Comment by wsor4035** [comment]
> Should you also add i3 stylization to the filter injector formspecs?

🤦‍♂️ missed that, added https://github.com/mt-mods/pipeworks/pull/22/commits/7dfa021272909c42ee728aec9b2d25f54d4f3630

**Comment by fluxionary** [comment]
> > Should you also add i3 stylization to the filter injector formspecs?
> 
> man_facepalming missed them, thanks for pointing them out, added [7dfa021](https://github.com/mt-mods/pipeworks/commit/7dfa021272909c42ee728aec9b2d25f54d4f3630)

screenshot_20220416_112555

misplaced button in the stackwise injector

**Comment by wsor4035** [comment]
> > > Should you also add i3 stylization to the filter injector formspecs?
> > 
> > 
> > man_facepalming missed them, thanks for pointing them out, added [7dfa021](https://github.com/mt-mods/pipeworks/commit/7dfa021272909c42ee728aec9b2d25f54d4f3630)
> 
> screenshot_20220416_112555
> 
> misplaced button in the stackwise injector

what game and mods?

**Comment by wsor4035** [comment]
> > > Should you also add i3 stylization to the filter injector formspecs?
> > 
> > 
> > man_facepalming missed them, thanks for pointing them out, added [7dfa021](https://github.com/mt-mods/pipeworks/commit/7dfa021272909c42ee728aec9b2d25f54d4f3630)
> 
> screenshot_20220416_112555
> 
> misplaced button in the stackwise injector

fixed https://github.com/mt-mods/pipeworks/pull/22/commits/7426e11104f6aa639e36288fbc61780436200cda

File: teleport_tube.lua [hunk 1]
```diff
@@ -143,17 +143,19 @@ local function update_meta(meta, can_receive)
...
+		"size["..size.."]"..
+		pipeworks.fs_helpers.get_prepends(size) ..
+		"image[0.5,o;1,1;pipeworks_teleport_tube_inv.png]"..
+		"label[1.5,0.5;"..S("Teleporting Tube").."]"..
+		"field[0.5,1.6;4.6,0.75;channel;"..S("Channel")..";${channel}]"..
+		"button[4.8,1.6;1.5,0.75;set_channel;"..S("Set").."]"..
+		"label[7.0,0.5;"..S("Receive").."]"..
+		"image_button[7.0,0.75;1,0.6;pipeworks_button_" .. cr_state .. ".png;cr" .. (can_receive and 0 or 1) .. ";;;false;pipeworks_button_interm.png]"..
+		"button_exit[6.3,1.6;2,0.75;close;"..S("Close").."]"..
+		"label[0.5,2.7;"..itext.."]")
```
[the remark this answers is not in the document]
Comment (fluxionary) [reply teleport_tube.lua]: screenshot_20220416_114501

**Comment by fluxionary** [comment]
how it looks on YL:

screenshot_20220416_115043

how it looks now, after the last commit ([37eef73](https://github.com/mt-mods/pipeworks/commit/37eef736950f038e54fd74d11a5d905fb43abdad))

screenshot_20220416_115059

File: teleport_tube.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (wsor4035) [reply teleport_tube.lua]: outdated

**Pull request merged**